src/main/kotlin/uk/gov/justice/digital/hmpps/hmppstemplatepackagename/config/WebClientConfiguration.kt

```diff
@@ -0,0 +1,82 @@
+package uk.gov.justice.digital.hmpps.hmppstemplatepackagename.config
+
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.http.client.reactive.ReactorClientHttpConnector
+import org.springframework.security.oauth2.client.AuthorizedClientServiceOAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientManager
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientProviderBuilder
+import org.springframework.security.oauth2.client.OAuth2AuthorizedClientService
+import org.springframework.security.oauth2.client.registration.ClientRegistrationRepository
+import org.springframework.security.oauth2.client.web.reactive.function.client.ServletOAuth2AuthorizedClientExchangeFilterFunction
+import org.springframework.web.reactive.function.client.WebClient
+import reactor.netty.http.client.HttpClient
+import java.time.Duration
+import kotlin.apply as kotlinApply
+
+@Configuration
+class WebClientConfiguration(
+  @Value("\${api.base.url.oauth}") val oauthApiBaseUri: String,
+  @Value("\${api.health-timeout:2s}") val healthTimeout: Duration,
+  @Value("\${api.timeout:90s}") val timeout: Duration,
+) {
+  @Bean
+  fun oauthApiHealthWebClient(builder: WebClient.Builder): WebClient = builder.healthWebClient(oauthApiBaseUri, healthTimeout)
+
+  /**
+   * TODO
+   * Once you have a client registration defined in properties `spring.security.client.registration` then you'll
+   *  need to uncomment this @Bean and create both a health and authorized web client.
```

Comment:
I've got an idea around generating all web clients in the library `hmpps-kotlin-spring-boot-starter` by reading  the client registrations from configuration and automatically creating web clients for each. But this will have to do for now.